File: Support/Templates/cs/Files/ReadOnlyList.cs
using System;
using System.Collections.Generic;
using Csla;

namespace Templates
{
  [Serializable]
  public class ReadOnlyList : 
    ReadOnlyListBase<ReadOnlyList, ReadOnlyChild>
  {
    #region Authorization Rules

    private static void AddObjectAuthorizationRules()
    {
      // TODO: add authorization rules
      //AuthorizationRules.AllowGet(typeof(ReadOnlyList), "Role");
    }

    #endregion

    #region Factory Methods

    public static ReadOnlyList GetReadOnlyList(string filter)
    {
      return DataPortal.Fetch<ReadOnlyList>(filter);
    }

    #endregion

    #region Data Access

    private void DataPortal_Fetch(string criteria)
    {
      RaiseListChangedEvents = false;
      IsReadOnly = false;
      // TODO: load values
      object objectData = null;
      foreach (var child in (List<object>)objectData)
        Add(ReadOnlyChild.GetReadOnlyChild(child));
      IsReadOnly = true;
      RaiseListChangedEvents = true;
    }

    #endregion
  }
}

File: Support/Templates/vb/Files/ReadOnlyList.vb
Imports System
Imports System.Collections.Generic
Imports Csla
Imports Csla.Security

<Serializable()>
Public Class ReadOnlyList
  Inherits ReadOnlyListBase(Of ReadOnlyList, ReadOnlyChild)

  Private Shared Sub AddObjectAuthorizationRules()
    ' TODO: add authorization rules 
    ' AuthorizationRules.AllowGet(GetType(ReadOnlyList), "Role")
  End Sub

  <Fetch>
  Private Sub Fetch(ByVal id As String)
    RaiseListChangedEvents = False
    IsReadOnly = False
    ' TODO: load values 
    Dim objectData As Object = Nothing
    For Each child As Object In DirectCast(objectData, List(Of Object))
      Add(DataPortal.FetchChild(Of ReadOnlyChild)(child))
    Next
    IsReadOnly = True
    RaiseListChangedEvents = True
  End Sub

End Class
